core/src/main/java/com/google/cloud/sql/ApplicationDefaultCredentialFactory.java

```diff
@@ -16,18 +16,15 @@
 
 package com.google.cloud.sql;
 
-import com.google.api.client.http.HttpRequestInitializer;
 import com.google.api.services.sqladmin.SQLAdminScopes;
-import com.google.auth.http.HttpCredentialsAdapter;
 import com.google.auth.oauth2.GoogleCredentials;
 import java.io.IOException;
 import java.util.Arrays;
 
 /** This class creates a HttpRequestInitializer from Application Default Credentials. */
 public class ApplicationDefaultCredentialFactory implements CredentialFactory {
-
   @Override
-  public HttpRequestInitializer create() {
+  public GoogleCredentials createGoogleCredentials() {
```

Comment:
This allows us to get rid of a lot of casting logic and tests to ensure casting is implemented correctly.